das/daser.go

```diff
@@ -263,3 +284,52 @@ func (d *DASer) sampleHeader(ctx context.Context, h *header.ExtendedHeader) erro
 
 	return nil
 }
+
+// SampleRoutineState reports the current state of the
+// DASer's main sampling routine.
+func (d *DASer) SampleRoutineState() RoutineState {
+	d.state.sampleLk.RLock()
+	state := d.state.sample
+	d.state.sampleLk.RUnlock()
+	return state
+}
+
+func (d *DASer) updateSampleState(h *header.ExtendedHeader, err error) {
+	height := uint64(h.Height)
+
+	d.state.sampleLk.Lock()
+	defer d.state.sampleLk.Unlock()
```

Comment:
Is there a reason to use `defer` here but not in the other functions?

Reply:
https://github.com/celestiaorg/celestia-node/pull/848/commits/821d5108f178fa8e84d98cf348514cbdc01b76fa done